integration-tests/README.md

```diff
@@ -0,0 +1,197 @@
+
+Since tests require internal services, the tenant and managed namespaces should remain as configured.
+
+## Running Tests
+
+### Basic Usage
+
+To run a specific test suite:
+
+```bash
+./run-test.sh <test-suite-name>
+```
+
+Example:
+```bash
+./run-test.sh fbc-release
+```
+
+### Command Line Options
+
+- **`--skip-cleanup`** or **`-sc`** - Skip cleanup operations after test completion (useful for debugging)
+- **`--no-cve`** or **`-nocve`** - Skip CVE simulation (where applicable)
+
+### Examples
+
+```bash
+# Run fbc-release test
+./run-test.sh fbc-release
+
+# Run test with debugging (skip cleanup)
+./run-test.sh fbc-release --skip-cleanup
+
+# Run collectors test without CVE simulation
+./run-test.sh collectors --no-cve
+```
+
+## Debugging
+
+### Debug Options
+
+When debugging test failures, use the `--skip-cleanup` option to preserve resources for examination:
+
+```bash
+./run-test.sh <test-suite-name> --skip-cleanup
+```
+
+### Manual Cleanup
+
+When debugging is complete, you can clean up resources using these scripts:
+
+- **`utils/cleanup-resources.sh`** - Cleans up Kubernetes resources
+- **`scripts/delete-branches.sh`** - Cleans up GitHub branches
+
+## Secret Management
+
+### Viewing Secrets
+
+To view encrypted secrets:
+
+```bash
+ansible-vault decrypt vault/tenant-secrets.yaml --output "/tmp/tenant-secrets.yaml" --vault-password-file <vault password file>
+```
+
+### Updating Secrets
+
+To update encrypted secrets:
+
+1. Decrypt the vault file:
+   ```bash
+   ansible-vault decrypt vault/tenant-secrets.yaml --output "/tmp/tenant-secrets.yaml" --vault-password-file <vault password file>
+   ```
+
+2. Edit the decrypted file:
+   ```bash
+   vi /tmp/tenant-secrets.yaml
+   ```
+
+3. Re-encrypt the file:
+   ```bash
+   ansible-vault encrypt /tmp/tenant-secrets.yaml --output "vault/tenant-secrets.yaml" --vault-password-file <vault password file>
+   ```
+
+4. Remove the temporary file:
+   ```bash
+   rm /tmp/tenant-secrets.yaml
+   ```
+
+## Test Execution Flow
+
+The integration tests follow this general workflow:
+
+1. **Environment Setup** - Load test-specific configuration and validate required variables
+2. **Secret Decryption** - Decrypt and apply required secrets to the cluster
+3. **GitHub Operations** - Create branches, make commits, and manage pull requests
+4. **Kubernetes Resources** - Create and manage namespaces, applications, components, and releases
+5. **Pipeline Execution** - Monitor Konflux Components and Tekton PipelineRuns
+6. **Verification** - Validate Release custom resources and pipeline outcomes
+7. **Cleanup** - Remove created resources (unless `--skip-cleanup` is specified)
+
+## CI/CD Integration
+
+These integration tests are automatically executed in CI/CD pipelines:
+
+- **Pull Request Triggers** - Tests run when changes are made to relevant pipeline files or the `integration-tests/` directory
+- **E2E Pipeline** - Uses `integration-tests/pipelines/e2e-tests-staging-pipeline.yaml`
```

Comment:
This can be resolved without action, but I don't even know what the difference between this one and the one below it is. Does anyone?

Reply:
@jinqi7

Reply:
e2e-tests-staging-pipeline.yaml is the integration pipeline for running e2e test cases written in bash.  
konflux-e2e-tests-pipeline.yaml is the integration pipeline for running the whole test suite of release-pipelines in e2e-tests repo. 
And we have two more pipelines -
e2e-tests-kind-pipeline.yaml pipeline is the integration pipeline for running one of the E2E test case (in e2e-tests repo)  which require a new cluster. 
e2e-tests-stage-pipeline.yaml pipeline is the integration pipeline for running one of the E2E test case (in e2e-tests repo)  which is using stage cluster.

Reply:
thanks. I knew about the latter 2 but not the first 2